ddp-workspace/projects/ddp-dsm-ui/src/app/dashboard-statistics/components/date-range/date-range.component.ts

```diff
@@ -0,0 +1,137 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output
+} from '@angular/core';
+import {AbstractControl, FormControl, FormGroup, ValidationErrors, Validators} from '@angular/forms';
+import {DateRangeModel} from '../../models/DateRange.model';
+import {Subject} from 'rxjs';
+import {takeUntil, auditTime} from 'rxjs/operators';
+import {DatePipe} from '@angular/common';
+import {Global} from '../../../globals/globals';
+
+@Component({
+  selector: 'app-date-range',
+  templateUrl: 'date-range.component.html',
+  styleUrls: ['date-range.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateRangeComponent implements OnInit, OnDestroy {
+  /**
+   * @MAIN Main Date Range Object
+   */
+  public readonly dateRangeForm: FormGroup = this.generateFormGroup;
+
+  private readonly destroyed$: Subject<void> = new Subject<void>();
+
+  constructor(private datePipe: DatePipe) {
+  }
+
+  @Input('activeDates') activeDatesOnInit: DateRangeModel;
+
+  @Input('disabled') set disabledState(isDisabled: boolean) {
+    typeof isDisabled === 'boolean' && this.disableOrEnable(isDisabled);
+  }
+
+  @Output() dateChanged = new EventEmitter<DateRangeModel>();
+
+
+  ngOnInit(): void {
+    this.initDateRangeForm();
+    this.listenToValueChangesAndEmit();
+  }
+
+  ngOnDestroy(): void {
+    this.destroyed$.next();
+  }
+
+  public getEntries(object: object): [string, any][] {
+    return Object.entries(object);
+  }
```

Comment:
The method does not make sense, imo.
Just use `Object.entries(your_object)`

Reply:
How to call that in the template?

Reply:
of course, one way was to create other pipe, but I didn't go further

Reply:
e.g. use [KeyValue pipe](https://angular.io/api/common/KeyValuePipe) in a template.
But in this particular case I guess you should simplify the error messages array which is dislpayed.

Reply:
yes, I tried this pipe as well, but in our structure didn't work properly, and didn't try to adjust it further. I was thinking about this as well, also how critical this can be but, of course, I will work on this.
Let's leave it open. I will let you know when I update the error messages approach.

Reply:
adjusted!